domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/AnalyticsControllerTest.kt

```diff
@@ -98,7 +100,9 @@ class AnalyticsControllerTest {
 
   @Before
   fun setUp() {
-    networkConnectionUtil = NetworkConnectionUtil(ApplicationProvider.getApplicationContext())
+    debugNetworkConnectionUtil = DebugNetworkConnectionUtil(
+      ProdNetworkConnectionUtil(ApplicationProvider.getApplicationContext())
+    )
```

Comment:
This seems wrong--why are we overwriting an injected field? We generally shouldn't do that (and, I think, this is just being ignored anyway since setUpTestApplicationComponent will overwrite this value when it calls inject(this)).

Reply:
Actually, it was previously initializing the `NetworkConnectionUtil`, so I did it for debug util as well. But, I did not realise that previously it was not having an `@Inject` constructor. I have removed these initializations now. PTAL. Thanks.